src/program_executor.cpp

```diff
@@ -139,6 +139,17 @@ void ProgramExecutionServer::Execute(
       return;
     }
   }
+
+  moveit_msgs::CollisionObject surface;
+  surface.header.frame_id = robot_state_.config.base_link();
```

Comment:
Most likely, we don't need to specify the frame ID when removing a collision object. Remove this line and test this change again after making all the other changes.